ntbs-service/Services/AlertService.cs

```diff
@@ -118,24 +115,42 @@ public async Task DismissAllOpenAlertsForNotification(int notificationId)
 
             foreach (var alert in alerts)
             {
+                // notification may have a duplicate alert to be dismissed on multiple other records
                 if (alert is DataQualityPotentialDuplicateAlert duplicateAlert)
                 {
-                    var duplicate =
-                        await _notificationRepository.GetNotificationForAlertCreationAsync(duplicateAlert.DuplicateId);
-
-                    if (duplicate.NotificationStatus.IsOpen())
-                    {
-                        continue;
-                    }
+                    await DismissDuplicationAlertsOnDuplicateRecord(notificationId, duplicateAlert.DuplicateId);
                 }
-
-                alert.ClosureDate = DateTime.Now;
-                alert.AlertStatus = AlertStatus.Closed;
+                
+                SetClosedAndAddClosureDate(alert);
             }
 
             await _alertRepository.SaveAlertChangesAsync();
         }
 
+        private async Task DismissDuplicationAlertsOnDuplicateRecord(int notificationId, int duplicateAlertId)
+        {
+            var duplicateRecordAlerts = await _alertRepository.GetAllOpenAlertsByNotificationId(duplicateAlertId);
+            if (!(duplicateRecordAlerts is null))
+            {
+                foreach (
+                    var alert in duplicateRecordAlerts
+                        .Where(al => al is DataQualityPotentialDuplicateAlert duplicateAlert && 
+                                     duplicateAlert.DuplicateId == notificationId)
+                         )
+                {
```

Comment:
Sorry I don't think I was clear enough in my previous comment, that's my mistake! I think this where clause should be on line 132. So something like: 
```suggestion
                var duplicateRecordAlerts = (await _alertRepository.GetAllOpenAlertsByNotificationId(duplicateAlertId))
                    .Where(al => al is DataQualityPotentialDuplicateAlert duplicateAlert && duplicateAlert.DuplicateId == notificationId));
```

Reply:
Ah I see - that makes sense! I tried something like this in the first instance but it wouldn't let me do a LINQ on a Task - turns out I needed was some brackets. Thanks!

Reply:
My personal preference with `await` would be to avoid wrapping it in brackets, but set it to an intermediary variable instead. It's a very minor thing, but I find it a little clearer. So something like:
```
var openAlertsOnDuplicateNotification = await _alertRepository.GetAllOpenAlertsByNotificationId(duplicateAlertId);
var duplicateAlertsOnDuplicateNotification = openAlerts
    .OfType<DataQualityPotentialDuplicateAlert>()
    .Where(a => a.DuplicateId == notificationId);
```

Reply:
One other thing is that `duplicateAlertId` seems like a bad variable name, because it appears to be a notification ID, not an alert ID.